lib/policies/proxy/proxy.js

```diff
@@ -50,23 +43,37 @@ module.exports = function (params, config) {
     }
   });
 
-  // single `url` property takes precedence over `urls` array
+  let balancer;
   if (endpoint.url) {
-    return function proxyHandler (req, res) {
-      logger.debug(`proxying to ${endpoint.url}, ${req.method} ${req.url}`);
-      proxy.web(req, res, proxyOptions);
-    };
+    // single `url` property takes precedence over `urls` array
+    balancer = createStrategy('static', proxyOptions, [endpoint.url]);
+  } else {
+    const strategy = params.strategy || 'round-robin';
+    balancer = createStrategy(strategy, proxyOptions, endpoint.urls);
   }
+  return function proxyHandler (req, res) {
+    const targetUrl = balancer.nextTarget();
+    proxyOptions.target = Object.assign(proxyOptions.target, targetUrl);
 
-  // multiple urls will load balance, defaulting to round-robin
-  const strategy = params.strategy || 'round-robin';
-  const balancer = createStrategy(strategy, proxy, proxyOptions, endpoint.urls);
-
-  return function balancerProxyHandler (req, res) {
-    return balancer.proxy(req, res);
+    logger.debug(`proxying to ${targetUrl}, ${req.method} ${req.url}`);
+    prepareHeaders(params, proxyOptions, req.egContext);
+    proxy.web(req, res, proxyOptions);
   };
+
+  // multiple urls will load balance, defaulting to round-robin
 };
 
+function prepareHeaders (params, proxyOptions, egContext) {
+  proxyOptions.headers = proxyOptions.headers || {};
+  // Default headers always sent to downstream
+  // TODO: allow configuration
+  if (egContext.requestId) {
+    proxyOptions.headers['eg-request-id'] = egContext.requestId;
+  }
+  proxyOptions.headers['eg-consumer-id'] = egContext.consumer && egContext.consumer.id
+  ? egContext.consumer.id : 'anonymous';
```

Comment:
Or — we could not send the header at all if the consumer is not specified.

Reply:
equivalent approaches. 
let's send it for now. we can make username configurable or provide option to disable it in future. 

I like sending the header to explicitly show that user has not passed auth